jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -366,6 +369,19 @@ def add_data(self, data, color=None, alpha=None, **layer_state):
         if len(self.layers) == 0:
             reset_plot_axes = True
         else:
+            # Check if the new data flux unit is actually compatible since flux not linked.
+            try:
+                uc.to_unit(data, data.find_component_id("flux"), [1, 1],
+                           u.Unit(self.state.y_display_unit))  # Error if incompatible
+            except Exception as err:
+                # Raising exception here introduces a dirty state that messes up next load_data
+                # but not raising exception also causes weird behavior unless we remove the data
+                # completely.
+                self.session.hub.broadcast(SnackbarMessage(
+                    f"Failed to load {data.label}, so removed it: {repr(err)}",
+                    sender=self, color='error'))
```

Comment:
```suggestion
                    f"Failed to load {data.label}: {repr(err)}", sender=self, color='error'))
```
Removing the data from the data collection is backend stuff that the user doesn't know about, so I don't think we need that part of this message. As far as they know, if it failed to load it shouldn't be in the data collection in the first place.

Reply:
Well, I added "so removed it" because there is a green "data is loaded" text for the same data right before that happens. With this revelation, does your suggestion still stand?